chain/chain/src/validate.rs

```diff
@@ -160,15 +162,15 @@ pub fn validate_chunk_with_chunk_extra(
         return Err(ErrorKind::InvalidBalanceBurnt.into());
     }
 
-    let receipt_response = chain_store.get_outgoing_receipts_for_shard(
+    let outgoing_receipts = chain_store.get_outgoing_receipts_for_shard(
+        runtime_adapter,
         *prev_block_hash,
         chunk_header.shard_id(),
-        prev_chunk_header.height_included(),
+        prev_chunk_height_included,
     )?;
     let outgoing_receipts_hashes = {
-        let epoch_id = runtime_adapter.get_epoch_id_from_prev_block(prev_block_hash)?;
-        let shard_layout = runtime_adapter.get_shard_layout(&epoch_id)?;
-        runtime_adapter.build_receipts_hashes(&receipt_response.1, &shard_layout)
+        let shard_layout = runtime_adapter.get_shard_layout_from_prev_block(prev_block_hash)?;
+        runtime_adapter.build_receipts_hashes(&outgoing_receipts, &shard_layout)
     };
     let (outgoing_receipts_root, _) = merklize(&outgoing_receipts_hashes);
 
```

Comment:
I think this validation is not fully correct. If I understand it correctly, for every split shard it does the same thing -- it tries to get the receipts from before the split and validate that against the outgoing receipt from last epoch before the layout changes. However, there seems to be no validation that the split itself is correct

Reply:
I'm not sure that I understand your question. The `get_outgoing_receipts_for_shard` function will return receipts after the split. Say shard changes after block `prev_hash` and block `hash`, then get_outgoing_receitpts_for_shard` function returns the outgoing receipts for shard `shard_id` in block `hash`